### PR TITLE
refactor: Cleanup state_test.go file in did Exchange

### LINKED ISSUE #679: Cleanup state_test.go file in did Exchange

### PR DESCRIPTION
**Pull request by talwinder50**
closes #679

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: pkg/didcomm/protocol/didexchange/states.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func (ctx *context) prepareAckConnectionRecord(payload []byte) (*ConnectionRecor
 	ack := &model.Ack{}
 	err := json.Unmarshal(payload, ack)
 	if err != nil {
-		return nil, err
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/states.go]: Should this be a wrapped error rather than %s?

Likewise below?

Comment (talwinder50) [reply pkg/didcomm/protocol/didexchange/states.go]: yes updated

File: pkg/didcomm/protocol/didexchange/states_test.go [hunk 2]
```diff
@@ -678,7 +549,7 @@ func TestVerifySignature(t *testing.T) {
 		require.NoError(t, err)
 
 		cs := &ConnectionSignature{
-			Type:       "https://didcomm.org/signature/1.0/ed25519Sha512_single",
+			Type:       "https://didcomm.org;spec/signature/1.0/ed25519Sha512_single",
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/states_test.go]: there is no ";spec"

**Comment by troyronda** [comment]
This PR will need a rebase.

File: pkg/didcomm/protocol/didexchange/states_test.go (hunk 2 above)
Comment (talwinder50) [reply pkg/didcomm/protocol/didexchange/states_test.go]: yea there is no spec removed

**Comment by talwinder50** [comment]
> This PR will need a rebase.

done

File: pkg/didcomm/protocol/didexchange/states_test.go [hunk 3]
```diff
@@ -1008,6 +897,87 @@ func createDIDDocWithKey(pub string) *diddoc.Doc {
...
+}
+
+func saveConnectionRecord(newDidDoc *diddoc.Doc, ctx *context) (*Response, error) {
+	response, err := createResponse(newDidDoc, generateRandomID(), ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	connRec := &ConnectionRecord{State: (&responded{}).Name(), ThreadID: response.Thread.ID,
+		ConnectionID: "123"}
```
Comment (soluchok) [inline pkg/didcomm/protocol/didexchange/states_test.go]: nit: next time please split long structure like this (one field in one line)
```
connRec := &ConnectionRecord{
    State: (&responded{}).Name(), 
    ThreadID: response.Thread.ID,
    ConnectionID: "123",
}
```

File: pkg/didcomm/protocol/didexchange/states.go [hunk 4]
```diff
@@ -173,12 +173,12 @@ func (s *requested) ExecuteInbound(msg *stateMachineMsg, thid string, ctx *conte
...
 		err := json.Unmarshal(msg.payload, invitation)
 		if err != nil {
-			return nil, nil, nil, fmt.Errorf("unmarshalling failed: %s", err)
+			return nil, nil, nil, fmt.Errorf("unmarshalling failed: %w", err)
 		}
 
 		action, connRecord, err := ctx.handleInboundInvitation(invitation, thid, msg.connRecord)
 		if err != nil {
-			return nil, nil, nil, fmt.Errorf("handle inbound invitation failed: %s", err)
+			return nil, nil, nil, fmt.Errorf("handle inbound invitation failed: %w", err)
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/didexchange/states.go]: We don't need to add `failed` to a wrapped error messages. 

Same comment for similar messages as well.

File: pkg/didcomm/protocol/didexchange/states_test.go (hunk 2 above)
Comment (troyronda) [reply pkg/didcomm/protocol/didexchange/states_test.go]: I still see it?

File: pkg/didcomm/protocol/didexchange/states_test.go (hunk 3 above)
Comment (talwinder50) [reply pkg/didcomm/protocol/didexchange/states_test.go]: done

File: pkg/didcomm/protocol/didexchange/states.go (hunk 4 above)
Comment (talwinder50) [reply pkg/didcomm/protocol/didexchange/states.go]: done

File: pkg/didcomm/protocol/didexchange/states_test.go (hunk 2 above)
Comment (talwinder50) [reply pkg/didcomm/protocol/didexchange/states_test.go]: just pushed the patch should not be there anymore

**Pull request merged**